xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-ui/src/main/resources/RTFrontend/Ajax.xml

```diff
@@ -74,11 +74,23 @@
   #if ("$!{request.convertHTML}" == '1')
     #set ($xdom = $services.rendering.parse($currentContent, 'xhtml/1.0'))
     #set ($currentContent = $services.rendering.render($xdom, $nextDocument.syntax.toIdString()))
+    ## FIXME: This requires programming rights.
+    #set ($discard = $xcontext.setDoc($nextDocument.document))
```

Comment:
Why don't you use `RTFrontend.Ajax` as template on the actual document to get the correct context document?

Reply:
@michitux you mean as sheet right? i.e. make the request to the edited document with ``?sheet=RTFrontend.Ajax``

Reply:
I removed the Ajax service. Getting rid of these rights issues.
Thanks.